projects/rocprofiler-systems/source/lib/core/trace_cache/cache_manager.cpp

```diff
@@ -189,6 +188,32 @@ cache_manager::post_process_bulk()
                 thread.join();
             }
         }
+
+        ROCPROFSYS_PRINT("Removing temporary files...\n");
+
+        auto remove_if_exists = [](const std::string& fname) {
+            std::ifstream file(fname);
+            if(file.is_open())
+            {
+                file.close();
+                if(std::remove(fname.c_str()) == 0)
+                {
+                    ROCPROFSYS_DEBUG("Removed file: %s\n", fname.c_str());
+                }
+                else
+                {
+                    ROCPROFSYS_WARNING(0, "Failed to remove file: %s\n", fname.c_str());
+                }
```

Comment:
Maybe some additional error checking on the return code.

```suggestion
                int result = std::remove(fname.c_str());

                if(result == 0)
                {
                    ROCPROFSYS_DEBUG("Removed file: %s\n", fname.c_str());
                }
                else if (errno == ENOENT)
                {
                    // File does not exist, so it was already removed
                    ROCPROFSYS_DEBUG("File does not exist: %s\n", fname.c_str());
                }
                else
                {
                    // Actual error - permission denied or other error
                    ROCPROFSYS_WARNING(0, "Failed to remove file: %s (errno: %d - %s)\n", 
                              fname.c_str(), errno, std::strerror(errno));
                }
```